### DISCUSSION TITLE
Edge on macOS

### DISCUSSION THREAD

**Discussion by payinez** [Acidham/chromium-hist-bookmarks#27] [Q&A] [answered]
Hello, 

I added this workflow to Alfred, but it still says no Bookmark found! 

I'm a big beginner, so I apologize if I'm doing something wrong. Could you help me?

thanks
P

**Comment by Acidham** [comment]
what browser are installed on your computer?

**Reply by payinez** [reply]
Microsoft Edge on macOS 13

**Reply by Acidham** [reply]
1. did you enable edge in workflow configuration?

2. can you please check if this file is available?

~/Library/Application Support/Microsoft Edge/Default/Bookmarks

**Reply by payinez** [reply]
1. yes, workflow is enable
2. I don't see any such folder. do I have to create it and put bm.html in it? Or should it have been created automatically with the edge installation?

**Comment by Acidham** [comment]
I don't mean enabled, I meant enable EDGE in "Configure Workflow". There you can find checkboxes for all the browsers.

**Reply by payinez** [reply]
Yes 
<img width="1359" alt="alfred" src="https://user-images.githubusercontent.com/120648267/208373174-e975e071-ec29-479e-813d-1d7831324f82.png">

**Comment by Acidham** [comment]
Can you please check where EDGE is storing the DEFAULT profile. Please send screenshot of the dirs under `~/Library/Application Support/Microsoft Edge/`

**Reply by payinez** [reply]
I can only see the path to Edge when the mode is on show/Hide hidden files. 
![Library - Edge](https://user-images.githubusercontent.com/120648267/208911274-ab2a14c7-5b20-4b01-b803-4454a56ad035.png)

**Comment by Acidham** [comment]
> I can only see the path to Edge when the mode is on show/Hide hidden files.

Yes that's normal, forgot to mention it.

I see now the issue. The Workflow does not support profiles and uses Default profile which is stored under `Default` folder. In your screenshot there is ONLY `Profile 2` and now `Default` which is strange. `Default` is, as far I know, always being created. 

As written before, the workflow is not supporting profiles, but you can tweak the path in the script. 

1. Open workflow folder by right-clicking in Alfred workflow → `Open in Finder`
2. Open `chrom_history.py` in your text editor and change line number 24 from `"edge": "Library/Application Support/Microsoft Edge/Default/History",` to `"edge": "Library/Application Support/Microsoft Edge/Profile 2/History",`
3. Save the file.

for Bookmark search follow steps above and edit the file `chrom_bookmarks.py` line number 25 from `"edge": 'Library/Application Support/Microsoft Edge/Default/Bookmarks',` to `"edge": 'Library/Application Support/Microsoft Edge/Profile 2/Bookmarks',`

**Reply by payinez** [reply]
small success, history works but bookmarks still not. if I check all browsers in config workflow, after typing bm even the message searches bookmarks disappears. 

shouldn't I add the ending bookmarks.bak or .msbak?
![bookmarks](https://user-images.githubusercontent.com/120648267/209131528-5b711fea-cdb9-4512-8436-cd6faeca7d56.png)

**Comment by Acidham** [comment]
Nope, these are backup files. I can see `Bookmarks` file. Can you please open Alfred workflow and open Debugger (small bug) top right. Then execute the workflow and let me know the error.

**Reply by payinez** [reply]
![bug](https://user-images.githubusercontent.com/120648267/209135689-7f52974f-bd97-440b-922f-9e8740183643.png)

**Comment by Acidham** [comment, marked as answer]
1. chrome bookmarks are also not found, please check this path as well. 
2. You potentially changed the path for Edge and add a leading `/`. Please remove that. The path should look like the same as for Chrome.

**Reply by payinez** [reply]
YES! `/` that was the problem. probably during various research I accidentally added /.

Thank you so much for your help and your time!

**Comment by mihajlovicn** [comment]
@Acidham , thank you very much for this great plugin. ;)

I experience a performance issue when searching for Edge based bookmarks - please see the attached log output:

```
10:39:56.038] Chromium Bookmarks and History Search[Script Filter] Queuing argument ''
[10:39:57.237] Chromium Bookmarks and History Search[Script Filter] Queuing argument 'stg3'
[10:40:14.746] Chromium Bookmarks and History Search[Script Filter] Script with argv '' finished
[10:40:14.749] STDERR: Chromium Bookmarks and History Search[Script Filter] PYTHON VERSION: 3.9.6 (default, Oct 18 2022, 12:41:40) 
[Clang 14.0.0 (clang-1400.0.29.202)]
/Users/s3462/Library/Application Support/Chromium/Default/Bookmarks → NOT found
/Users/s3462/Library/Application Support/Microsoft Edge/Default/Bookmarks → found
[10:40:14.750] Chromium Bookmarks and History Search[Script Filter] {
    "items": [
        {
            "title": "'AADSTS50126' \"invalid_grant\" Error validating credentials due to invalid username or password - Microsoft Q&A",
            "subtitle": "https://docs.microsoft.com/en-us/answers/questions/38035/aadsts50126-invalid-gra",
            "arg": "https://docs.microsoft.com/en-us/answers/questions/38035/aadsts50126-invalid-grant-error-validating-credent.html",
            "quicklookurl": "https://docs.microsoft.com/en-us/answers/questions/38035/aadsts50126-invalid-grant
```

I'm not sure why it takes almost more than 15 seconds to load and show the bookmarks. This makes the plugin almost unusable for me. Let me know what additonal information you need for analysis support.

**Comment by Acidham** [comment]
The logs are looking good. How many bookmarks have you stored in Edge?
